utils/snapshotHub.ts

```diff
@@ -0,0 +1,319 @@
+   */
+  snapshot: number;
+} & SnapshotCoreProposalData;
+
+export type SnapshotVoteData = {
+  /**
+   * Core data about the vote.
+   */
+  payload: {
+    /**
+     * The choice's index
+     */
+    choice: number;
+    /**
+     * The proposal's IPFS hash
+     */
+    proposalIpfsHash: string;
+    /**
+     * Stringifiable metadata for the proposal, e.g. `private: 1`
+     */
+    metadata: SnapshotCoreProposalPayloadData["metadata"];
+  };
+} & SnapshotCoreProposalData;
+
+export type SnapshotMessage = {
+  name: SnapshotCoreProposalPayloadData["name"];
+  body: SnapshotCoreProposalPayloadData["body"];
+  metadata: SnapshotCoreProposalPayloadData["metadata"];
+  token: SnapshotCoreProposalData["token"];
+  space: SnapshotCoreProposalData["space"];
+  actionId: SnapshotCoreProposalData["actionId"];
+  chainId: SnapshotCoreProposalData["chainId"];
+  verifyingContract: SnapshotCoreProposalData["verifyingContract"];
+};
+
+export type SnapshotMessageProposal = {
+  /**
+   * Ethereum block to use (e.g. latest).
+   */
+  snapshot: SnapshotProposalData["snapshot"];
+  /**
+   * Duration in seconds of the vote.
+   */
+  votingTimeSeconds: number;
+} & SnapshotMessage;
+
+export type SnapshotMessageVote = {
+  /**
+   * Choice to submit for the vote.
+   */
+  choice: VoteChoices;
+  /**
+   * Stringifiable metadata for the proposal, e.g. `private: 1`
+   */
+  metadata: {
+    /**
+     * Allows for a different address other than the `address` voting.
+     * In many cases this will be the same as the `address` field.
+     *
+     * e.g. Calculating membership voting power, but a delegate address was used for `address`.
+     */
+    memberAddress: string;
+  } & Record<string, any>;
+  /**
+   * Ethereum network ID. For ERC712 signature verification.
+   */
+  chainId: number;
+};
+
+export type SnapshotVoteProposal = {
+  actionId: SnapshotCoreProposalData["actionId"];
+  proposalIpfsHash: SnapshotVoteData["payload"]["proposalIpfsHash"];
+  space: SnapshotCoreProposalData["space"];
+  token: SnapshotCoreProposalData["token"];
+  verifyingContract: SnapshotCoreProposalData["verifyingContract"];
+};
+
+const VOTE_CHOICES: CoreProposalVoteChoices = [VoteChoices.Yes, VoteChoices.No];
+
+const addSecondsTimestamp = (
+  tsSeconds: string,
+  secondsToAdd: number
+): string => {
+  return (parseInt(tsSeconds) + secondsToAdd).toFixed();
+};
+
+const getTimestampSeconds: () => string = () => (Date.now() / 1e3).toFixed();
+
+// @note The snapshot-hub API does not accept falsy choices like index `0`.
+const getVoteChoiceIndex = (choice: VoteChoices) =>
+  VOTE_CHOICES.findIndex((c) => c === choice) + 1;
+
+export const buildDraftMessage = async (
+  message: SnapshotMessage,
+  snapshotHubURL: string
+): Promise<SnapshotDraftData> => {
+  try {
+    const { data } = await getApiStatus(snapshotHubURL);
+
+    return {
+      payload: {
+        body: message.body,
+        choices: VOTE_CHOICES,
+        metadata: message.metadata,
```

Comment:
I think it's OK, because it's on `lao-backends/lao-server` we have the custom queries, correct? If that's the case, I am OK with each app having control over its own metadata, but not forcing this library to accommodate multiple app's data  
they rely upon.

There may be a way at a later time to find a better way to add "structure" to `metadata` per `space`?